torch/csrc/jit/passes/utils/check_alias_annotation.cpp

```diff
@@ -69,6 +62,20 @@ bool deepEquals(const IValue& lhs, const IValue& rhs) {
     return lhs.toTensor().equal(rhs.toTensor());
   }
 
+  if (lhs.isTensorList() && rhs.isTensorList()) {
```

Comment:
Do the changes in this PR fail without adding this code? I'm surprised that the `lhs == rhs` at the bottom of this function doesn't properly compare TensorLists and Tensors already. From my understanding `==`-ing IValues should already perform the logic implemented here

Reply:
I believe without this code lhs == rhs will return a tensor of booleans which then fails since the function expects the return value to be a single bool. I will double check that this is the case.

Reply:
@Lilyjjo The problem with lhs == rhs is that it will compare the Tensors in the lists using the == instead of .equal() which will return a Tensor instead of bool.